File: cs365/hw5/Stack/Stack.cpp
// Stack.cpp
// Mitchell Wade
// Feb. 17, 2015
// This program is implementation for a stack in C++.

// CONSTRUCTOR
template <class E, int MAX_SIZE>
Stack<E, MAX_SIZE>::Stack(int maxSize): maxCap(maxSize) {}

// PUSH
template <class E, int MAX_SIZE>
bool Stack<E, MAX_SIZE>::push(E value) {
    
    if (data.size() == maxCap)
        return false;
    else {
        data.push_back(value);
        return true;
    }
}

// POP
template <class E, int MAX_SIZE>
E Stack<E, MAX_SIZE>::pop() {
    
    E value = data.back();
    data.pop_back();
    
    return value;
}

// ISEMPTY
template <class E, int MAX_SIZE>
bool Stack<E, MAX_SIZE>::isEmpty() {
    
    return data.empty();
}

File: cs365/hw5/Stack/Stack/Stack.java
// Stack.java
// Mitchell Wade
// Feb. 17, 2015
// This program is implementation for a stack in Java.

package Stack;

import java.util.*;

public class Stack <T>{

    Vector <T> v = new Vector <T>();
    private int MaxCap = 0;
    private final static int MAX_SIZE = 100;
    private T value;

    // DEFAULT CONSTUCTOR
    public Stack() {
        MaxCap = MAX_SIZE;
    }

    // SPECIFIED CONSTRUCTOR
    public Stack(int maxSize) {
        MaxCap = maxSize;
    }

    // PUSH
    public boolean push(T value) {
        
        if (v.size() == MaxCap) {
            return false;
        } else {
            v.add(value);
            return true;
        }
    }

    // POP
    public T pop() {

        value = v.remove(0);
        return value;

    }

    // ISEMPTY
    public boolean isEmpty() {
        return v.isEmpty();
    }
}
